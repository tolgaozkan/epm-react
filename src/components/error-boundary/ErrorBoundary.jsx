import PropTypes from 'prop-types';
import React from 'react';

export default class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  componentDidCatch(error, info) {
    // Display fallback UI
    this.setState({ hasError: true });

    console.error(error, info);
  }

  render() {
    if (this.state.hasError) {
      // You can render any custom fallback UI
      return <h1>Something went wrong.</h1>;
    }
    return this.props.children;
  }
}

ErrorBoundary.propTypes = {
  children: PropTypes.object.isRequired,
};
