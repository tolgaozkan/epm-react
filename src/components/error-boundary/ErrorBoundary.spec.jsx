import * as React from 'react';
import { create as render } from 'react-test-renderer';

import ErrorBoundary from './ErrorBoundary';

it('renders  error boundary', () => {
  const children = (<div />);

  const result = render(<ErrorBoundary children={children} />);
  const snap = result.toJSON();

  snap.state = { hasError: true };

  expect(snap).toMatchSnapshot();
});
