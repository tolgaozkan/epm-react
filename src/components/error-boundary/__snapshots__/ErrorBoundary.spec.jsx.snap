// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders error boundary 1`] = `<div />`;
